File: Client-Leger/app/src/main/java/com/zouatene/colorimage/model/drawelement/Shape.kt
package com.zouatene.colorimage.model.drawelement

data class Shape(
    override var id: String,
    override var z: Int,
    override var matrix: IMatrix?,
    override var isSelected: Boolean,
    var initialPoint: Point,
    var color: String?,
    var finalPoint: Point,
    var strokeColor: String?,
    override var strokeWidth: Float,
    var text: String?,
    var textColor: String?
): VectorObject()

File: Client lourd/client/src/app/interfaces/drawing/shape.ts
import { Point } from "./point";
import { VectorObject } from "./vectorObject";

export interface Shape extends VectorObject {
    strokeColor: string,
    initialPoint: Point,
    finalPoint: Point,
    text: string,
    textColor: string
}

export interface TypedShape extends Shape {
    type : string
}

export function instanceOfLine(object: any): object is Shape {
    return 'strokeColor' in object && 'initialPoint' in object && 'finalPoint' in object;
}